core/lib/dr_tools.h

```diff
@@ -48,6 +48,12 @@ DR_API
  * \warning This context cannot be used as the drcontext for a thread
  * running under DR control!  It is only for standalone programs that
  * wish to use DR as a library of disassembly, etc. routines.
+ * \warning This context is not fully thread-safe as it stores
+ * the current ISA mode DR uses in a global variable.
```

Comment:
There are other variables in dcontext_t like encode_state, decode_state to which the same concern applies; and there may be more in future. Better to say that "... as it stores some state (like isa_mode) that may be prone to data races".

Reply:
Ah, yes, and the comments in the code point at #1595 filed long ago on this dcontext issue.  So for AArch32, decoding and encoding are problematic, even if not switching ISA modes.

Reply:
Done. Also mentioned AArch32 parallel encoding/decoding issues.